Eastern-Research-Group/expert-query
# Feature/54 etl cron job

## Related Issues:
* [EQ-54](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-54)

## Main Changes:
* Set up a cron job for the ATTAINS ETL process.
  * For testing, this is set to run every 30 minutes.
* Created a table to log the process.
* Created a table to store current schemas.
* Set up logic to rotate the new schema into use after creation.
* Set up logic to remove old schemas.
  * For testing, this is executed on schemas older than one day.

## Steps To Test:
1. In the `database/docker/postgres` directory, start the test database with the command `docker compose up`
2. In the `etl` directory, start the ETL process with the command `npm run start`
3. Navigate to [localhost:8080](http://localhost:8080)
4.  Login with credentials "PostgreSQL", "postgres", "postgres", "example", "expert_query" for the System, Server, Username, Password, & Database fields, respectively
5. Confirm that a "logging" schema is created with the tables "etl_log" and "etl_schemas", and that the tables are populated with current values
6. Logout, and log back in with credentials "PostgreSQL", "postgres", "eq", "expert", "expert_query" for the System, Server, Username, Password, & Database fields, respectively
7. Confirm that a schema is created with the label "schema_{timestamp}", and that it is populated with a placeholder table
8. Check that logs are being output properly to the terminal
9. Check the logs to confirm that the process is being run every 30 minutes
